src/lib/storage/disk_analyzer.rb

```diff
@@ -0,0 +1,368 @@
+              next
+            else
+              return true if installation_volume?(partition)
+            end
+          end
+          return false # if we get here, there is a partition table.
+        rescue RuntimeError => ex
+          log.info("CAUGHT exception: #{ex} for #{disk}")
+        end
+
+        # Check if there is a filesystem directly on the disk (without partition table).
+        # This is very common for installation media such as USB sticks.
+        installation_volume?(disk)
+      end
+
+      # Check if a volume (a partition or a disk without a partition table) is
+      # our installation medium - the medium we just booted and started the
+      # installation from.
+      #
+      # The method to achieve this is to try to mount the filesystem and check
+      # if there is a file /control.xml and compare its content to the
+      # /control.xml in the inst-sys. We cannot simply compare device
+      # minor/major IDs since the inst-sys is in a RAM disk (copied from the
+      # installation medium).
+      #
+      def installation_volume?(vol)
+        log.info("Checking if #{vol.name} is an installation volume")
+        is_inst = mount_and_check(vol) { |mp| installation_volume_check(mp) }
+        log.info("#{vol} is installation medium") if is_inst
+        is_inst
+      end
+
+      # Check if the volume mounted at 'mount_point' is an installation volume.
+      # This is a separate method so it can be redefined in unit tests.
+      #
+      # @return 'true' if it is an installation volume, 'false' if not.
+      #
+      def installation_volume_check(mount_point)
+        check_file = "/control.xml"
+        mount_point += "/" unless mount_point.end_with?("/")
+        return false unless File.exist?(mount_point + check_file)
+        FileUtils.identical?(check_file, mount_point + check_file)
+      end
+
+      # Mount a volume, perform the check given in 'block' while mounted, and
+      # then unmount. The block will get the mount point of the volume as a
+      # parameter.
+      #
+      # @return the return value of 'block' or 'nil' if there was an error.
+      #
+      def mount_and_check(vol, &block)
+        raise ArgumentError, "Code block required" unless block_given?
+        mount_point = "/mnt" # FIXME
+        begin
+          # check if we have a filesystem
+          # return false unless vol.filesystem
+          mount(vol.name, mount_point)
+          check_result = block.call(mount_point)
+          umount(mount_point)
+          check_result
+        rescue RuntimeError => ex
+          log.error("CAUGHT exception: #{ex} for #{vol}")
+          nil
+        end
+      end
+
+      # Return an array of the device names of the specified block devices
+      # (::Storage::Disk, ::Storage::Partition, ...).
+      #
+      # @param disks [Array<BlkDev>]
+      # @return [Array<string>] names
+      #
+      def dev_names(blk_devices)
+        blk_devices.map { |x| x.to_s }
```

Comment:
it depends on background, if you get familiar with it, it really looks nature.
Compare how you read code:
- your one: "Map each blk_device to x and use result of call to_s on x" . Not bad, quite short, just mapping is not necessary

- suggestion: "Map each blk to its to_s result". Shorter and remove unnecessary mapping, so less brain capacity needed t do mapping.